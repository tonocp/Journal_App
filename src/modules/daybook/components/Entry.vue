<template>
  <div class="entry-container mb-3 pointer p-2">
    <!-- Título -->
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">15</span>
      <span class="mx-1 fs-5">Diciembre</span>
      <span class="mx-2 fw-light">2021, miércoles</span>
    </div>

    <div class="entry-description">
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Eius possimus fugiat ipsa, libero aut illo, quibusdam
      obcaecati voluptates odit, voluptatum soluta minus asperiores quis? Autem necessitatibus voluptatem officia sit
      provident.
    </div>
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }

  .entry-description {
    font-size: 12px;
  }
}
</style>
