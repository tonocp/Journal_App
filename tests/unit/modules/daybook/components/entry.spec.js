import { shallowMount } from "@vue/test-utils";

import Entry from "@/modules/daybook/components/Entry.vue";
import { journalState } from "../../../mock-data/test-journal-state";

describe("Pruebas en el Entry Component", () => {
  const mockRouter = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(Entry, {
    props: { entry: journalState.entries[0] },
    global: { mocks: { $router: mockRouter } },
  });

  test("Debe de hacer match con el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Debe redireccionar al hacer click en el entry-container", () => {
    const entryContainer = wrapper.find(".entry-container");
    entryContainer.trigger("click");

    expect(mockRouter.push).toHaveBeenCalledWith({ name: "entry", params: { id: journalState.entries[0].id } });
  });

  test("Pruebas en las propiedades computadas", () => {
    expect(wrapper.vm.day).toBe(16);
    expect(wrapper.vm.month).toBe("Diciembre");
    expect(wrapper.vm.yearDay).toBe("2021, Jueves");
  });
});
