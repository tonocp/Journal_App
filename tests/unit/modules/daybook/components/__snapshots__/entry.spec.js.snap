// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry Component Debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Título -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">16</span><span class="mx-1 fs-5">Diciembre</span><span class="mx-2 fw-light">2021, Jueves</span></div><!-- Descripción -->
  <div class="entry-description">Hola mundo desde mock data</div>
</div>
`;
